utils/rector/src/Rector/Renaming/RenameClassNameRector.php

```diff
@@ -71,16 +71,18 @@ public function getNodeTypes(): array
 	 */
 	public function refactor(Node $node): ?Node
 	{
-		if ($node instanceof Class_) {
-			$oldName = $node->name->toLowerString();
-			$names = explode('_', $oldName);
+        $oldName = $node->name->toLowerString();
+        if (! str_contains($oldName, '_')) {
+            return null;
+        }
 
-			// Take name parts, capitalize the first letter and return a string.
-			$newName = implode('', array_map(fn($namePart) => ucfirst($namePart), $names));
+        $names = explode('_', $oldName);
 
-			return $this->classRenamer->renameNode($node, [$oldName => $newName]);
```

Comment:
In this case, `classRename` is too heavy service for class renaming. It also depends on few internals that are related to original `RenameClassRector` rule.